MCTS.pyx

```diff
@@ -224,6 +236,8 @@ class MCTS():
             # leaf node
             self.Ps[s], v = self.nnet.predict(canonicalBoard)
             valids = self.game.getValidMoves(canonicalBoard, 1)
+            if s == self.init_board:
```

Comment:
The noise should be added to the root node of a search, not the init board of the game. This wouldn't do much.

Reply:
Here I defined the root node of the search as being equivalent to the starting state as string, stored in self.init_board.

Reply:
Yes, but that is not the correct root for the entire game.
On the very first turn, the init board is the root, but not afterwards. The root should be the current game state. So after a player has made a move, the root has to change to match the new game state.